LLMS-compatibility blogpost (#45)
upsun/devcenter: dev/content/posts/llms-introduction.md

---
title: "Upsun Docs is Now llms.txt compatible!"
subtitle: ""
date: 2025-02-13T10:00:00+00:00
image: /images/llms-introduction/llms-introduction.webp
image_alt: "Image generated by AI"
icon: tutorial
featured: true
author:
  - flovntp

sidebar:
    exclude: true
type: post

description: |
  We’re thrilled to announce that [Upsun Documentation](https://docs.upsun.com) is now fully compatible with [llms.txt proposed standard](https://llmstxt.org/)!
    
tags:
  - llms
  - ai
categories:
  - featured
  - releases
math: false

---

![Main image showing an AI analysing Markdown files](/images/llms-introduction/llms-introduction.webp "Image generated by AI")

We’re thrilled to announce that [Upsun Documentation](https://docs.upsun.com) is now fully compatible with [llms.txt proposed standard](https://llmstxt.org/)! 

This update brings enhanced integrations, better automation, and more flexibility to your documentation workflows.

> ## What is llms.txt and Why it matters

Reviewer: ```suggestion
## What is llms.txt and why does it matter?
```

> With the enthusiasm for AI assistants, many folks are looking for these assistants to provide technical information about your product.

Reviewer: ```suggestion
With the growing enthusiasm for AI, many folks are looking for AI-powered assistants to provide technical information about their product.
```

After years of remarkable achievements through web 2.0 and its famous ``robots.txt``, ``security.txt``, and ``humans.txt``,
a new standard has been proposed to the web ecosystem and will soon become essential for the web: [``llms.txt``](https://llmstxt.org/).

`llms.txt` was conceived by [Jeremy Howard](https://github.com/jph00), co-founder of [Answer.AI](https://answerai.pro/), to address a fundamental challenge in AI-human interaction.

When AI assistants attempt to process standard web pages, they struggle with non-essential elements like navigation menus, scripts, and styling.

> These elements consume valuable context space without contributing to the actual content understanding.

Reviewer: ```suggestion
These elements consume valuable contextual space without contributing to an actual understanding of the content from the AIs perspective.
```

> `llms.txt` provides an elegant solution: it delivers precisely curated information in a format that AI systems can efficiently process and understand.

Reviewer: ```suggestion
`llms.txt` provides an elegant solution by delivering precisely curated information in a format that AI systems can efficiently process.
```

> By exposing publicly such ``llms.txt`` and ``llms-full.txt`` files, AI Assistants will be able to easily parse your documentation and ingest technical information about your product, to then help your end users understand how to use it.

Reviewer: ```suggestion
By making ``llms.txt`` and ``llms-full.txt`` files publicly available, AI Assistants will now be able to easily parse documentation and ingest technical information about a product to help your end users understand how to use it.
```

> ## Upsun Docs is Now `llms.txt`-Compatible

Reviewer: ```suggestion
## Upsun Docs is now `llms.txt`-compatible
```

> Upsun Docs is now `llms.txt`-compatible by exposing a [llms.txt](https://docs.upsun.com/llms.txt) and a [llms-full.txt](https://docs.upsun.com/llms-full.txt) files.

Reviewer: ```suggestion
Upsun Docs is now `llms.txt`-compatible, thanks to the public availability of the [llms.txt](https://docs.upsun.com/llms.txt) and [llms-full.txt](https://docs.upsun.com/llms-full.txt) files.
```

> These two files can be used in external LLMs or AI assistants by feeding them with these files.

Reviewer: ```suggestion
These two files can be fed into external LLMs or AI assistants to accurately parse Upsun Docs content.
```

Additionally, every documentation page is now available in Markdown format, with an added **View Markdown version** button on the right menu of each page. 
![View Markdown version button](/images/llms-introduction/view-md-version-button.png)
This allows users to easily access and export the raw Markdown format for their own needs.

> ## What’s New in Upsun Docs?

Reviewer: ```suggestion
## What’s new in Upsun Docs?
```

With ``llms.txt`` compatibility, Upsun Docs introduces:

> - **Seamless Integration with AI Assistants**: LLMs compatibility allows AI tools to interact with Upsun Docs more efficiently.

Reviewer: ```suggestion
- **Seamless integration with AI Assistants**: LLMs compatibility allows AI tools to interact with Upsun Docs more efficiently.
```

> - **Content Structuring for Learning Systems**: If you’re using Upsun Docs within an LMS environment, you’ll experience a smoother, more structured experience.

Reviewer: ```suggestion
- **Content structuring for learning systems**: The use of Upsun Docs within an LMS environment will guarantee a smoother, more structured experience.
```

> - **Markdown Export for Every Page**: Easily download and reuse documentation content with a dedicated Markdown version button.

Reviewer: ```suggestion
- **Markdown export for every page**: Easily download and reuse documentation content with a dedicated Markdown version button.
```

> ## How `llms.txt` Benefits You

Reviewer: ```suggestion
## How `llms.txt` benefits you
```

Whether you’re a developer, technical writer, or enterprise team, ``llms.txt`` compatibility enhances your documentation experience:

### For Developers

- **Train your AI Helper** with ``llms.txt`` and ``llms-full.txt`` to improve your Upsun experience.
- Enable **real-time content** updates within your ecosystem.

### For Technical Writers

- **Simplify** content structuring for better readability and navigation.
- **Optimize searchability** with improved metadata indexing.
- **Export Markdown versions easily** for documentation portability.

### For Teams & Businesses

- Integrate documentation **seamlessly with training platforms**.
- Enhance **collaboration and knowledge** sharing across teams.
- **Use Markdown for easy integration into wikis and other platforms**.

> ## How to Get Started

Reviewer: ```suggestion
## How to get started
```

> Because Upsun is new (released beginning of 2024), AI systems like Cursor, ChatGPT, Claude, and Copilot won’t give useful answers about it. 

Reviewer: ```suggestion
Upsun was recently released (at the start of 2024), therefore AI systems like Cursor, ChatGPT, Claude, and Copilot are yet to provide in-depth knowledge about the product. 
```

> To fix that problem, we’ve provided two LLM-friendly files that teaches them how to use Upsun. 

Reviewer: ```suggestion
To fix that problem, we’ve provided two LLM-friendly files that teach them how to use Upsun. 
```

> To use them, add this links for your AI helper to use:

Reviewer: ```suggestion
To use them, add these links to your AI helper:
```

- [/llms.txt](https://docs.upsun.com/llms.txt)
- [/llms-full.txt](https://docs.upsun.com/llms-full.txt)

> If you use [Cursor](https://www.cursor.com/), type `@doc` then choose “_Add new doc_”, and use the 2 `llms` links above. 

Reviewer: ```suggestion
{{< callout type="blue" >}}
If you use [Cursor](https://www.cursor.com/), type `@doc` then choose “_Add new doc_”, and use the two `llms` links above. 
{{< /callout >}}
```

> ## Transparency vs. AI Scraping: The Upsun Approach

Reviewer: ```suggestion
## Transparency vs. AI Scraping: The Upsun approach
```

Recently, some websites have started implementing "**tarpits**" to trap AI scrapers that ignore `robots.txt`, 
as described in this [Ars Technica article](https://arstechnica.com/tech-policy/2025/01/ai-haters-build-tarpits-to-trap-and-trick-ai-scrapers-that-ignore-robots-txt/). 
The goal is to prevent unauthorized use of content in AI models by feeding them incorrect data or slowing down their requests.

> At [Upsun](https://www.upsun.com), we take a different approach: instead of completely blocking AI access, we provide a **structured and official version of our documentation via ``llms.txt`` proposed standard**. 

Reviewer: ```suggestion
At [Upsun](https://www.upsun.com), we take a different approach: instead of completely blocking AI access, we provide a **structured and official version of our documentation via ``llms.txt`` proposed standards**. 
```

By exposing publicly such ``llms.txt`` and ``llms-full.txt`` files, AI Assistants will be able to easily parse your documentation and ingest technical information about your product, to then help your end users understand how to use it.

## Upsun Docs is Now `llms.txt`-Compatible

Upsun Docs is now `llms.txt`-compatible by exposing a [llms.txt](https://docs.upsun.com/llms.txt) and a [llms-full.txt](https://docs.upsun.com/llms-full.txt) files.
These two files can be used in external LLMs or AI assistants by feeding them with these files.

Additionally, every documentation page is now available in Markdown format, with an added **View Markdown version** button on the right menu of each page. 
![View Markdown version button](/images/llms-introduction/view-md-version-button.png)
This allows users to easily access and export the raw Markdown format for their own needs.

## What’s New in Upsun Docs?

With ``llms.txt`` compatibility, Upsun Docs introduces:

- **Seamless Integration with AI Assistants**: LLMs compatibility allows AI tools to interact with Upsun Docs more efficiently.

> - **Enhanced API for Automation**: Developers can now streamline content updates and integrations with ease.

Reviewer 2: Can we elaborate on this point? I don't quite understand this bullet.
